blockdigest/digest.go

```diff
@@ -59,6 +59,17 @@ func (digest Digest) Cmp(difficulty *big.Int) int {
 	return result.SetBytes(bigEndian[:]).Cmp(difficulty)
 }
 
+func (d Digest) SmallerDigestThan(target Digest) bool {
+	for i := len(d) - 1; i >= 0; i-- {
+		if d[i] == target[i] {
```

Comment:
Seem redundant check. just need to return true in case `d[i] < target[i]`

Reply:
yes, this can be changed